dask/backends.py

```diff
@@ -119,7 +120,15 @@ def decorator(
 
             @wraps(fn)
             def wrapper(*args, **kwargs):
-                return getattr(self, dispatch_name)(*args, **kwargs)
+                func = getattr(self, dispatch_name)
+                try:
+                    return func(*args, **kwargs)
+                except Exception as e:
+                    raise RuntimeError(
```

Comment:
```suggestion
                    raise type(e)(
```
@jrbourbeau - Should we try to preserve the original exception type, instead of using `RuntimeError`? Otherwise, we will need to change tests that are expecting a different exception (like in [test_creation::test_arange](https://github.com/dask/dask/blob/0ab2a1b75f575d63e97279ee3e5c22ab46c0737f/dask/array/tests/test_creation.py#L226)).